codes/models/modules/DUF_arch.py

```diff
@@ -0,0 +1,682 @@
+        else:
+            x3 = torch.cat((x2, x3), 1)
+        return x3
+
+
+class DUF_16L_noBN(nn.Module):
+    ''' same structure as that in the paper
+    https://github.com/yhjo09/VSR-DUF'''
+
+    def __init__(self):
+        super(DUF_16L_noBN, self).__init__()
+        self.conv3d_1 = nn.Conv3d(3, 64, (1, 3, 3), stride=(1, 1, 1), padding=(0, 1, 1), bias=True)
+        self.dense_block_1 = DenseBlock_noBN(64, 64 // 2,
+                                             t_reduce=False)  # 64 + 32 * 3 = 160, T = 7
+        self.dense_block_2 = DenseBlock_noBN(160, 64 // 2,
+                                             t_reduce=True)  # 160 + 32 * 3 = 160, T = 1
+
+        self.conv3d_2 = nn.Conv3d(256, 256, (1, 3, 3), stride=(1, 1, 1), padding=(0, 1, 1),
+                                  bias=True)
+
+        self.conv3d_r1 = nn.Conv3d(256, 256, (1, 1, 1), stride=(1, 1, 1), padding=(0, 0, 0),
+                                   bias=True)
+        self.conv3d_r2 = nn.Conv3d(256, 3 * 16, (1, 1, 1), stride=(1, 1, 1), padding=(0, 0, 0),
+                                   bias=True)
+
+        self.conv3d_f1 = nn.Conv3d(256, 512, (1, 1, 1), stride=(1, 1, 1), padding=(0, 0, 0),
+                                   bias=True)
+        self.conv3d_f2 = nn.Conv3d(512, 1 * 5 * 5 * 16, (1, 1, 1), stride=(1, 1, 1),
+                                   padding=(0, 0, 0), bias=True)
+
+        self.dynamic_filter = DynamicUpsamplingFilter_3C((1, 5, 5))
+
+        self.lrelu = nn.LeakyReLU(negative_slope=0.1, inplace=True)
+        # initialization
+        _initialize_weights(self.conv3d_1, 0.1)
+        _initialize_weights(self.conv3d_2, 0.1)
+        _initialize_weights(self.conv3d_r1, 0.1)
+        _initialize_weights(self.conv3d_r2, 0.1)
+        _initialize_weights(self.conv3d_f1, 0.1)
+        _initialize_weights(self.conv3d_f2, 0.1)
+
+    def forward(self, x):
+        '''
+        x: [B, T, C, H, W], T = 7. reshape to [B, C, T, H, W] for Conv3D
+        Generate filters and image residual:
+        Fx: [B, 25, 16, H, W] for DynamicUpsamplingFilter_3C
+        Rx: [B, 3*16, 1, H, W]
+        '''
+        B, T, C, H, W = x.size()
+        x = x.permute(0, 2, 1, 3, 4)  # [B,C,T,H,W] for Conv3D
+        x_center = x[:, :, T // 2, :, :]
+        x = self.conv3d_1(x)
+        x = self.dense_block_1(x)
+        x = self.dense_block_2(x)  # reduce T to 1
+        x = self.lrelu(self.conv3d_2(x))
+
+        # image residual
+        Rx = self.conv3d_r2(self.lrelu(self.conv3d_r1(x)))  # [B, 3*16, 1, H, W]
+        # filter
+        Fx = self.conv3d_f2(self.lrelu(self.conv3d_f1(x)))  # [B, 25*16, 1, H, W]
+        Fx = F.softmax(Fx.view(B, 25, 16, H, W), dim=1)
+
+        # dynamic filter
+        out = self.dynamic_filter(x_center, Fx)  # [B, 3*R, H, W]
+        out += Rx.squeeze_(2)
+        out = F.pixel_shuffle(out, 4)  # [B, 3, H, W]
+        return out
+
+
+class DenseBlock_28L_A(nn.Module):
+    '''Dense block w/ BN
```

Comment:
For the class `class DenseBlock_28L_A`:
1. modify the comments to be more meaningful. explain the purpose of this class. why call it xxx_A
2. I like the codes of `52L`. Could it also be possible to change the 28L to the style of `52L`.